foxtrot-core/src/main/java/com/flipkart/foxtrot/core/exception/DocumentMissingException.java

```diff
@@ -0,0 +1,44 @@
+package com.flipkart.foxtrot.core.exception;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by rishabh.goyal on 18/12/15.
+ */
+public class DocumentMissingException extends FoxtrotException {
+
+    private String table;
+    private List<String> ids;
+
+    public DocumentMissingException(String table, List<String> ids) {
```

Comment:
You have create* functions in the base class, even if you move them out to another class (see comment on base class), these should be package local as only the create* functions will access this.